articles/introduction-to-r/index.md

```diff
@@ -0,0 +1,328 @@
+
+### RStudio
+
+Before we begin, we need to ensure that the R interpreter is installed on our local machine. In this article, we will focus on programming concepts. You can use online editors like [Repl.it](https://repl.it/languages/rlang) for this tutorial. For further information on installation, refer to this [article](https://techvidvan.com/tutorials/install-r/).
+
+The RStudio interface has four components that we make use of most. Below is a preview of what the RStudio interface looks like.
+
+![rstuido](rstudio.jpg)
+
+The four components are as follows:
+
+1. **Editor**: The editor is used to create R scripts. R scripts are executed using the **run** button or the console. The **run** button is present at the top of the RStudio window. To execute a file using the console, we need to go through the following steps:
+   1. Set the current working directory to the directory, including the R script. Use the command `setwd(dir)` to set the working directory to the directory of your choice. Use `getwd()` to get the current working directory
+   2. We use the `source` command to execute the script file of our choice. `source(file_name)` executes the entire file and gives the output in the console section. The `source(filename)` is typed and executed via the console, which we cover in the next point.
+   ![](R-Script.jpg)
+2. **Console and Terminal**: The **console** in R is similar to IDLE in Python. We can execute any command here. For this tutorial, you may use the **console**.
+   ![](console.jpg)
+3. **Environment Variables**: The **environment variables window** list all the variables created during program execution and their corresponding values. It is used to understand the dimensionality and data-types of the data being used by a program.
+   ![](environment.jpg)
+4. **Plots, Package manager, and Help Section**: The plots section generates any plots that have been specified in the R script. The **Package Manager** gives us a list of packages installed and lets us install packages if required.  Finally, the help section is a go-to for developers of all levels. The ability to go through official docs through the RStudio empowers programmers and saves them much time.
+   ![](plot.jpg)
+
+
+### R-Programming Basics
+
+In this article, we will be covering the following concepts:
+
+1. Variables, Data Types, and Operations
+2. Vectors
+3. Matrices
+4. Factors
+5. Lists
+
+#### Variables, Data Types and Operations
+
+R includes the following atomic data-types: logical, numeric, integer, complex, and character. Abstract data types such as lists, stacks, etc., are defined using these atomic data-types. Variables are memory locations that store values during the execution of a computer program. To define a variable in R, we use the reverse-arrow(<-) operator. Let us look at the code to declare variables.
```

Comment:
Data types = 2 separate word. Make sure to remove hyphen.
Please clarify atomic data types as the most fundamental data structures, and abstract data types as building blocks created using atomic data types.
Also, there are 6 atomic data types (you forgot raw).
"Let's look at the code sample below to learn how to declare variables"